src/main/java/seedu/dukeofbooks/data/book/Book.java

```diff
@@ -32,5 +33,25 @@ public Person getAuthor() {
     public void setAuthor(Person author) {
         this.author = author;
     }
-
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Book // instanceof handles nulls
+                && this.hasSameData((Book) other));
```

Comment:
This boolean expression is a bit too long? 
Suggests: 
``` java
if (other == this) {
  return true;
} else {  // if is not the same object
  other instanceof Book && hasSameData
}

Reply:
This is also for other `equals`.

Reply:
Was thinking that the expression was clear, but will make the changes for greater code readability.